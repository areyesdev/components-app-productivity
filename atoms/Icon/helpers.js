const ICONS_ASSETS_PATH = '/icons',
  iconMap = {
    arrowRight: 'arrow-right',
    angleDown: 'angle-down',
    iconPlus: 'icon-plus',
  },
  iconSize = {
    sm: 15,
    md: 26,
    lg: 40,
    xl: 40,
  }

export const mapType = (type) => `${ICONS_ASSETS_PATH}/${iconMap[type]}.svg`
export const mapSize = (size) => iconSize[size]
