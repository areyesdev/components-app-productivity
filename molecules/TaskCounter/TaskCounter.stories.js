import { TaskCounter, styles } from '.'

import { getTemplate } from '../../helpers/storybook'

const Template = getTemplate(TaskCounter, styles)

export default {
  title: 'Molecules/TaskCounter',
  component: TaskCounter,
  args: {
    title: 'Em Sméagol body.',
    current: 2,
  },
}

export const Default = Template.bind({})
export const Total = Template.bind({})
Total.args = { total: 10 }

export const Toggleable = Template.bind({})
Toggleable.args = {
  isToggleable: true,
  total: 10,
  children:
    "Fells defeated veins this. There and Back Again. A Hobbit's Tale. Loved lucky Istari difficult lay? Dale fellow's rioting. Blanket Misty Mountain started rival wriggles flatten convinced succumbed closing? Slug town fond pipe-weed flatten recognizes Goblins! Innards pouf admirable freely nothingness sleepless exactly afternoon cruelty Bofur? Craftsmen tested fair Alfrid gets figure words winter's field offer healed squash? These galumphing Grima tread mustn't seen melt ideal picked recover employment. Nazgûl alas luckiest?",
}
