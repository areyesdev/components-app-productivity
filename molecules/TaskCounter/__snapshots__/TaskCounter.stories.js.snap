// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/TaskCounter Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="task-counter"
  >
    <div
      className="container"
    >
      <p
        className="paragraph color-base size-md weight-medium"
      >
        Em Sméagol body.
      </p>
      <p
        className="paragraph color-base size-md weight-medium is-inline"
      >
        2
      </p>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 5,
        }
      }
    />
    <div
      className="divider"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TaskCounter Toggleable 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="accordion"
  >
    <div
      className="container"
      onClick={[Function]}
    >
      <div
        className="icon color-inverted size-sm background-muted"
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      >
        <svg
          height={25}
          viewBox="0 0 15 9"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="spacer horizontal"
        style={
          Object {
            "height": 15,
          }
        }
      />
      <p
        className="paragraph color-base size-md weight-medium"
      >
        Em Sméagol body.
      </p>
      <p
        className="paragraph color-base size-md weight-medium is-inline"
      >
        2/10
      </p>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 5,
        }
      }
    />
    <div
      className="divider"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TaskCounter Total 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="task-counter"
  >
    <div
      className="container"
    >
      <p
        className="paragraph color-base size-md weight-medium"
      >
        Em Sméagol body.
      </p>
      <p
        className="paragraph color-base size-md weight-medium is-inline"
      >
        2/10
      </p>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 5,
        }
      }
    />
    <div
      className="divider"
    />
  </div>
</div>
`;
